testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -805,6 +832,131 @@ fn stx_transfer_btc_integration_test() {
+
+    let mut btc_regtest_controller = BitcoinRegtestController::new(conf.clone(), None);
+    let http_origin = format!("http://{}", &conf.node.rpc_bind);
+
+    btc_regtest_controller.bootstrap_chain(201);
+
+    eprintln!("Chain bootstrapped...");
+
+    let mut run_loop = neon::RunLoop::new(conf.clone());
+    let blocks_processed = run_loop.get_blocks_processed_arc();
+
+    let channel = run_loop.get_coordinator_channel().unwrap();
+
+    thread::spawn(move || run_loop.start(None, 0));
+
+    // give the run loop some time to start up!
+    wait_for_runloop(&blocks_processed);
+
+    // first block wakes up the run loop
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // first block will hold our VRF registration
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // next block, issue a commit
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // let's figure out the current chain tip
+    let chain_tip = get_chain_tip(&http_origin);
+
+    // HACK: this test relies on manually inserting a bad microblock into the chain state.
+    //  this behavior is not guaranteed to continue to work like this, so at some point this
+    //  test will need to be updated to handle that.
+    {
+        let (mut chainstate, _) =
+            StacksChainState::open(false, conf.burnchain.chain_id, &conf.get_chainstate_path())
+                .unwrap();
+        let mut tx = chainstate.db_tx_begin().unwrap();
+
+        let (consensus_hash, stacks_block) = get_tip_anchored_block(&conf);
+
+        //        let tip_hash = StacksBlockId::new(&consensus_hash, &stacks_block.header.block_hash());
+
+        let ublock_privk =
+            find_microblock_privkey(&conf, &stacks_block.header.microblock_pubkey_hash, 1024)
+                .unwrap();
+
+        let garbage_tx = make_stacks_transfer_mblock_only(
+            &spender_sk,
+            0,
+            100,
+            &PrincipalData::from(StacksAddress::burn_address(false)),
+            1000,
+        );
+        let mut garbage_block = StacksMicroblock::first_unsigned(
+            &chain_tip.1,
+            vec![StacksTransaction::consensus_deserialize(&mut garbage_tx.as_slice()).unwrap()],
+        );
+        garbage_block.header.prev_block = BlockHeaderHash([3; 32]);
+        garbage_block.header.sequence = 1;
+        garbage_block.sign(&ublock_privk).unwrap();
+
+        eprintln!("Minting microblock at {}/{}", &chain_tip.0, &chain_tip.1);
+        StacksChainState::store_staging_microblock(
+            &mut tx,
+            &consensus_hash,
+            &stacks_block.header.block_hash(),
+            &garbage_block,
+        )
+        .unwrap();
+        tx.commit().unwrap();
+    }
+
+    thread::sleep(Duration::from_secs(30));
```

Comment:
Is this long enough to prevent the node from trying lots of RBF's?  Is there a way to measure or reliably detect when the RBF limit gets reached, or is that no longer a concern with PR 2452?

Reply:
The goal is to get it to do exactly that -- attempt tons of RBFs. Without the changes in this PR, the final assertion would fails with like `29 != 1` (i.e., the test miner would have issued 29 block commits).